src/pygama/dsp/build_dsp.py

```diff
@@ -128,36 +130,37 @@ def build_dsp(f_raw, f_dsp, dsp_config, lh5_tables=None, database=None,
     for tb in lh5_tables:
         # load primary table and build processing chain and output table
         tot_n_rows = raw_store.read_n_rows(tb, f_raw)
-        if n_max and n_max<tot_n_rows: tot_n_rows=n_max
+        if n_max and n_max < tot_n_rows:
+            tot_n_rows = n_max
 
         # if we have separate DSP files for each table, read them in here
         if chan_config is not None:
             f_config = chan_config[tb]
             with open(f_config) as config_file:
-                dsp_config = json.load(config_file, object_pairs_hook=OrderedDict)
-            print('Processing table:', tb, 'with DSP config file:\n  ', f_config)
+                dsp_config = json.load(config_file, object_pairs_hook=json.OrderedDict)
+            log.debug(f'processing table: {tb} with DSP config file {f_config}')
 
         if not isinstance(dsp_config, dict):
-            raise Exception('Error, dsp_config must be an dict')
+            raise RuntimeError('dsp_config for {tb} must be a dict')
 
         chan_name = tb.split('/')[0]
         db_dict = database.get(chan_name) if database else None
         tb_name = tb.replace('/raw', '/dsp')
 
         write_offset = 0
         raw_store.gimme_file(f_dsp, 'a')
-        if write_mode=='a' and lh5.ls(f_dsp, tb_name):
+        if write_mode == 'a' and lh5.ls(f_dsp, tb_name):
             write_offset = raw_store.read_n_rows(tb_name, f_dsp)
 
         # Main processing loop
-        print(f'Processing table: {tb} ...')
-        lh5_it = lh5.LH5Iterator(f_raw, tb, buffer_len = buffer_len)
+        lh5_it = lh5.LH5Iterator(f_raw, tb, buffer_len=buffer_len)
         proc_chain = None
         for lh5_in, start_row, n_rows in lh5_it:
             # Initialize
             if proc_chain is None:
-                proc_chain, lh5_it.field_mask, tb_out = build_processing_chain(lh5_in, dsp_config, db_dict, outputs, verbose, block_width)
-                tqdm_it = iter(tqdm_range(0, tot_n_rows, buffer_len, verbose))
+                proc_chain, lh5_it.field_mask, tb_out = build_processing_chain(lh5_in, dsp_config, db_dict, outputs, block_width)
+                progress_bar = tqdm(desc=f'Processing table {tb}', total=tot_n_rows,
```

Comment:
Should this be `if log.level <= logging.INFO` like the update below?
```suggestion
                if log.level <= logging.INFO:
                    progress_bar = tqdm(desc=f'Processing table {tb}', total=tot_n_rows,
```